src/promptflow-core/promptflow/_utils/async_utils.py

```diff
@@ -109,3 +111,29 @@ async def wrapper(*args, **kwargs):
             return await asyncio.get_event_loop().run_in_executor(executor, partial_func)
 
     return wrapper
+
+
+async def consume_from_queue_async(queue: Queue):
+    while True:
+        if queue.qsize() == 0:
+            #  If the queue is empty, wait for a while and try again.
+            await asyncio.sleep(0.1)
+            continue
+        item = queue.get_nowait()
+        if item is None:
+            break
+        yield item
+
+
+def consume_generator_and_put_to_queue(generator: GeneratorType, queue: Queue):
+    for item in generator:
+        queue.put(item)
+    queue.put(None)
+
+
+async def sync_generator_to_async(g: GeneratorType):
+    q = Queue()
+    pool = ThreadPoolExecutorWithContext(max_workers=1)
+    pool.submit(consume_generator_and_put_to_queue, g, q)
+    async for item in consume_from_queue_async(q):
```

Comment:
I suspect if this async for could guarantee the output sequence of.

For example currently in the queue there is two element 1,2
async for doesn't guarantee the output sequence is exactly 1,2   2,1 is also valid output.

Reply:
Updated with `next`